sydent/replication/pusher.py

```diff
@@ -72,10 +72,12 @@ def scheduledPush(self):
         peers = self.peerStore.getAllPeers()
 
         # Push to all peers in parallel
-        return defer.DeferredList([self._push_to_peer(p) for p in peers])
+        dl = []
+        for p in peers:
+            dl.append(defer.ensureDeferred(self._push_to_peer(p)))
+        return defer.DeferredList(dl)
 
-    @defer.inlineCallbacks
-    def _push_to_peer(self, p: "RemotePeer") -> Generator:
+    async def _push_to_peer(self, p: "RemotePeer") -> Generator:
```

Comment:
This no longer returns a generator. (It seems to return `NoReturn`.)

Reply:
When I change the return value to `NoReturn`, mypy highlights that there are a few return statements in this function, which all return nothing and seem to function to break out of execution, like here: https://github.com/matrix-org/sydent/blob/b798a834204afcb521cfd08fe6ee192827978dd5/sydent/replication/pusher.py#L92 Is it okay to let that error (`misc - Implicit return in function which does not return`) ride or is there a way to fix it?

Reply:
@H-Shay You should be able to use `None` instead of no return here then.